### PR TITLE
Update matchFilter to use getItems

### PR DESCRIPTION
**Pull request by timleslie**
This PR removes `matchFilter` and instead uses direct calls to `getItems` from `server-side-graphql-client`. It removes the need to do client side sorting and removes an unnecessary level of abstraction which makes these `test-fixtures` harder to understand.

### CODE REVIEW

File: packages/test-utils/lib/test-utils.js [hunk 1]
```diff
@@ -6,7 +6,6 @@ const { Keystone } = require('@keystonejs/keystone');
 const { GraphQLApp } = require('@keystonejs/app-graphql');
 const { KnexAdapter } = require('@keystonejs/adapter-knex');
 const { MongooseAdapter } = require('@keystonejs/adapter-mongoose');
-const { runCustomQuery } = require('@keystonejs/server-side-graphql-client');
```
Comment (timleslie) [inline packages/test-utils/lib/test-utils.js:9]: I believe this means we can remove the package dependency on `server-side-graphql-client`

File: .changeset/small-cups-protect.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@keystonejs/fields': minor
+---
+
+Updated `match` function to use `getItems` function from `server-side-graphql-client`.
```
Comment (timleslie) [inline .changeset/small-cups-protect.md:5]: The `fields` package doesn't need a `major` change, as it's only update its tests. We do however need a `major` changeset on `test-utils`, as we're removing a function from its public API.

Comment (singhArmani) [reply .changeset/small-cups-protect.md:5]: Yeah, I was kind of thinking on the same line. But wasn't sure.

File: .changeset/small-cups-protect.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@keystonejs/fields': patch
+---
+
+Updated `match` function to use `getItems` function from `server-side-graphql-client`.
```
Comment (timleslie) [inline .changeset/small-cups-protect.md:5]:
```suggestion
Updated `match` function in tests to use `getItems` function from `server-side-graphql-client`.
```

File: packages/test-utils/index.js [hunk 4]
```diff
@@ -4,7 +4,6 @@ const {
   graphqlRequest,
   authedGraphqlRequest,
   networkedGraphqlRequest,
-  matchFilter,
```
Comment (timleslie) [inline packages/test-utils/index.js:7]: Need to revert these changes.
